### ISSUE TITLE
Change from YML to TOML

### ISSUE THREAD

**Issue by emthompson-usgs** [usgs/groundmotion-processing#235]
Because of support for comments.

**Comment by smithj382** [comment]
I used ruamel.yaml package: https://pypi.org/project/ruamel.yaml/ to preserve comments for project creation.

The branch is viewable here: https://github.com/smithj382/groundmotion-processing/tree/keep_config_comments

I tested it on a new project and all comments from the production config file are persistent.

ruamel.yaml seems to be more actively maintained and suitable for this purpose (unless there is a decision made to switch from YAML type config to something else).

I will hold off on creating a pull request until I get some feedback suggesting this is an appropriate way forward for the config package. Please let me know what you think.

**Comment by emthompson-usgs** [comment]
After a bit of empirical testing, it seems that the section ordering in the conf is maintained in the resulting dict keys. So we should be able to convert to configobj with a structure that looks like this:
```
[processing]
    [[detrend]]
        detrending_method = demean

    [[compute_snr]]
        # Presense of this check says to do the signa-to-noise ratio check. 
        # Requires minimum SNR of `threshold` between `min_freq` and `max_freq` 
        # using Konno-Omachi smoothed spectra with `bandwidth` parameter.
        bandwidth = 20.0
        [[[check]]]
            threshold = 3.0
            min_freq = 'f0'  # Either a float or 'f0'. If 'f0', then the Brune
            # corner frequency of the earthquake will be used as the minimum
            # frequency for the SNR check.
            max_freq = 5.0
            [[[[f0_options]]]]
                stress_drop = 10  # bars
                shear_vel = 3.7   # km/s
                ceiling = 2.0     # Hz. If the computed f0 is greater than
                # ceiling, then the ceiling frequency will be used. This should
                # be less than max_freq.
                floor = 0.1       # Hz. If the computed f0 is less than the
                # floor, then the floor frequency will be used.
```

**Comment by emthompson-usgs** [comment]
Yes, thanks for pointing that out @jmfee-usgs. I had forgotten about that issue. We already have multiple "detrend" steps with different parameters.

**Comment by cbworden** [comment]
I think any of these approaches would probably work. If you're already using yaml, then @smithj382 might have the simplest solution. But if you are thinking of a refactor, maybe you could specify what a full config would ideally produce (without the comments) -- maybe as a Python structure or something, and then it would be clearer how to structure each alternative. Configobj can get ugly with nested brackets, but is really geared toward config files: checks everything, ranges, types, provides defaults, etc. It's more complicated to use, but does a lot that you would otherwise have to do manually. The other two, as I understand them, are far simpler to use, and basically turn the input file into a data structure, at which point you are on your own in terms of validation, etc. That may not be a problem if all of your functions do their own validation and provide default parameters, but it won't help you find problems in config files. Dunno. Your call. I have no horse in this race.
